lib/blocks.php

```diff
@@ -145,30 +145,21 @@ function gutenberg_serialize_block( $block ) {
 }
 
 /**
- * Renders a single block into a HTML string.
+ * Returns an array of the names of all registered dynamic block types.
  *
- * @since 1.9.0
- *
- * @param  array $block A single parsed block object.
- * @return string String of rendered HTML.
+ * @return array Array of dynamic block names.
  */
-function gutenberg_render_block( $block ) {
```

Comment:
I think we should keep both methods — with the full tree-parsing being only an option. (Say a REST endpoint that wants to serve the tree as JSON, etc.)

Reply:
>I think we should keep both methods — with the full tree-parsing being only an option. (Say a REST endpoint that wants to serve the tree as JSON, etc.)

The parsing class is left untouched, and in fact the `gutenberg_parse_blocks` helper function itself; this is what seemed to be the most value in what came to mind as needing a JSON tree of blocks. Do you think this specific function is still valuable, to generate the string output of a block object? All of the underlying tooling is still largely available, it just becomes:

```php
$html = $block_type->is_dynamic() ? $block_type->render( $block['attrs'] ) : $block['innerHTML'];
```